vm/runtime.ts

```diff
@@ -653,21 +654,25 @@ module J2ME {
   /** @const */ export var MIN_PRIORITY: number = 1;
   /** @const */ export var NORMAL_PRIORITY: number = 5;
 
+  /** @const */ export var ISOLATE_MIN_PRIORITY: number = 1;
+  /** @const */ export var ISOLATE_NORM_PRIORITY: number = 2;
+  /** @const */ export var ISOLATE_MAX_PRIORITY: number = 3;
+
   class PriorityQueue {
     private _top: number;
     private _queues: Context[][];
 
     constructor() {
-      this._top = MIN_PRIORITY;
+      this._top = MIN_PRIORITY + ISOLATE_MIN_PRIORITY;
       this._queues = [];
-      for (var i = MIN_PRIORITY; i <= MAX_PRIORITY; i++) {
+      for (var i = MIN_PRIORITY + ISOLATE_MIN_PRIORITY; i <= MAX_PRIORITY + ISOLATE_MAX_PRIORITY; i++) {
         this._queues[i] = [];
       }
     }
 
     enqueue(ctx: Context) {
-      var priority = ctx.getPriority();
-      release || assert(priority >= MIN_PRIORITY && priority <= MAX_PRIORITY,
+      var priority = ctx.getPriority() + ctx.runtime.priority;
```

Comment:
I tried it on this branch:

https://github.com/mykmelez/j2me.js/compare/mykmelez:prioritize-runtimes...mykmelez:prioritize-runtimes-through-isolate

But PriorityQueue.enqueue threw `TypeError: ctx.runtime.isolate is undefined` for some reason, so I had to check if it existed first.

Then it worked, but the numbers were 8, 9, 8, i.e. the button never animated, but the second screen appeared after that number of seconds, which is 2x slower. I'm not sure why the button didn't animate, but presumably the perf regression is because of the conditional.

If we wanted to ensure that it's possible to change the Isolate priority at dynamically, we might be able to update its priority in *Isolate.setPriority0*. We can't do that when the Isolate is initially created, because *Isolate.setPriority0* is called before *JVM.startIsolate*, which is what creates the runtime. But we could do it later. Nevertheless, I don't think there's any code that sets the Isolate's priority later, so I don't think we need to worry about this.